alibi_detect/utils/validate.py

```diff
@@ -34,6 +34,12 @@ def validate_config(cfg: dict, resolved: bool = False) -> dict:
     else:
         raise ValueError(f'Loading the {detector_name} detector from a config.toml is not yet supported.')
 
+    # Raise warning if config file already contains a version_warning
+    version_warning = cfg.get('version_warning', False)
```

Comment:
Is it enough to just save a boolean flag or perhaps it would be wiser to save the original version of the library (and `config_spec`?) into an immutable config field?

Reply:
I didn't think of that, it'll serve as nice record-keeping in the config too. Will add.

Reply:
@jklaise @mauicv I started to rework this to save an immutable `orig_version` and `orig_config_spec` field instead of `version_warning`. But after thinking about it more, I'm having second thoughts.

For an example: imagine we save a config file with `version=0.9.1`. Later we load it in version `0.10.0`, this would raise a warning as desired. However, if we saved the config again, we'd rewrite `version=0.10.0`. Therefore on subsequent loads we wouldn't be warned that there is potentially a problem. This is where `version_warning` comes in; it would be written to the config upon the first version mismatch, and then will cause a warning to be raised on subsequent loads regardless of `version`. 

Onto the proposed `orig_version`, I'm wondering if this extra info would just be confusing. In the above scenario, we'd write `orig_version=0.9.1`. A warning would be correctly raised on first load with `version=0.10.0`. However, the problem scenario is if we then saved a config, and loaded it back in with version 0.9.1 installed. This would still raise a warning since `version=0.10.0` in the config but the current version is 0.9.1. However, I think might lead to a false sense of security (or just be confusing) now we have `orig_version=0.9.1` and current installed version 0.9.1. 

TL;DR - what would this extra `orig_version` info actually give us? It seems to me that it might just lead to confusion...

Reply:
I agree with the sentiment. I think we might be missing a generally accepted pattern here as this problem would come up any time someone implements saving/loading functionality somewhere, perhaps it would be useful to briefly survey the problem.

Otherwise I agree that it might be less confusing to just stick with `version_warning`.